### ISSUE TITLE
Mutating Scala code yields code that doesn't compile

### ISSUE THREAD

**Issue by mthmulders** [stryker-mutator/stryker4s#148]
Note sure if I should file an issue about this, or if this is expected behavior...

### Summary

Running Stryker4S on [my application](https://github.com/mthmulders/hyperion) mutates some code into a non-compiling state.

### Stryker4s config

```hocon
stryker4s {
  base-dir = "app"
  log-level = debug
  mutate = [
    "src/main/scala/**/*.scala"
  ]
  reporters = [
    console
  ]
}
```

The original code is
```scala
def retrieveMeterReading(date: LocalDate): Future[Option[HistoricalMeterReading]] = {
  val query = meterReadings.filter(_.recordDate === date)
  db.run(query.result) map (_.headOption)
}
```

and it gets mutated into
```scala
def retrieveMeterReading(date: LocalDate): Future[Option[HistoricalMeterReading]] = {
  val query = sys.props.get("ACTIVE_MUTATION").orElse(sys.env.get("ACTIVE_MUTATION")) match {
    case Some("26") =>
      meterReadings.filterNot(_.recordDate === date)
    case _ =>
      meterReadings.filter(_.recordDate === date)
  }
  db.run(query.result) map _.headOption
}
```

The complete source file is [here](https://github.com/mthmulders/hyperion/blob/master/app/src/main/scala/hyperion/database/MeterReadingDAO.scala).

### Stryker4s environment

```
stryker4s 0.1.0+19-8c5511eb-SNAPSHOT
```

### Your Environment
| software             | version(s) |
| -------------------- | ---------- |
| Scala version        | 2.12.8 |
| Build tool & version | SBT 1.2.8 | 
| Operating System     | macOS 10.14.2  |
| Java Runtime | 1.8.0_202 |

**Comment by hugo-vrijswijk** [comment]
Hmm, that ain't right 🤔. This should've been fixed in #147. I'll have a closer look at it tomorrow

**Comment by Wmaarts** [comment]
Right, the fix we introduced in #147 doesn't work because this is different syntax.

We introduced a fix for i.e. `list map (add(_, 1))` to compile correctly after mutating.
But not if you do this `list map (add(_.someFunction))`.

Because then the Term.Placeholder (= underscore) is wrapped in a Term.Select together with the Term.Name (= `someFunction`). 

We don't check for this. :') Time to go back to Scalameta.
